app/models/work-item.ts

```diff
@@ -260,6 +260,37 @@ export async function getWorkItemsByJobId(
   };
 }
 
+/**
+ * Returns work items for a job step
+ * @param tx - the transaction to use for querying
+ * @param jobID - the job ID
+ * @param workflowStepIndex - the index of the workflow step
+ * @param currentPage - the page of work items to get
+ * @param perPage - number of results to include per page
+ * @param sortOrder - orderBy string (desc or asc)
+ *
+ * @returns A promise with the work items array
+ */
+export async function getWorkItemsByJobIdAndStepIndex(
+  tx: Transaction,
+  jobID: string,
+  workflowStepIndex: number,
+  currentPage = 0,
+  perPage = 10,
```

Comment:
I'm not sure if 10 is our normal default for paging, but since calling this function is internal code rather than some user request, and in the place we're using it now we're going to want to iterate through all of the items I think we'd want to use a larger page size. I'd either increase the default here to something like 100 or 500, or pass in a larger value when we call this function rather than using the default.

Reply:
Increased to 100, but that seems like an arbitrary number to me (as does 10) and will obviously use more memory than 10.